internal/provider/kubernetes/gateway.go

```diff
@@ -135,24 +166,30 @@ func (r *gatewayReconciler) Reconcile(ctx context.Context, request reconcile.Req
 		r.resources.Gateways.Delete(request.NamespacedName)
 	}
 
-	// Set the "Scheduled" condition to true for all accepted gateways.
-	for _, gw := range acceptedGateways {
+	// Set status conditions for all accepted gateways.
+	for i := range acceptedGateways {
+		gw := acceptedGateways[i]
+		// Get the status address of the Gateway's associated Service.
+		svc, err := r.serviceForGateway(ctx, &gw)
```

Comment:
we should probably loop across all gateways and update status as a best effort rather than bail out early

Reply:
OK, now the controller logs the error and continues.